src/main/java/seedu/pill/util/Storage.java

```diff
@@ -0,0 +1,157 @@
+        try {
+            File file = initializeFile();
+            FileWriter fw = new FileWriter(file);
+
+            for (String itemName : itemMap.items.keySet()) {
+                TreeSet<Item> itemSet = itemMap.items.get(itemName);
+                for (Item item : itemSet) {
+                    fw.write(item.getName() + SEPARATOR + item.getQuantity());
+
+                    if (item.getExpiryDate().isPresent()) {
+                        fw.write(SEPARATOR + item.getExpiryDate().get().toString());
+                    } else {
+                        fw.write(SEPARATOR + "");
+                    }
+
+                    fw.write(System.lineSeparator());
+                }
+            }
+
+            fw.close();
+        } catch (IOException e) {
+            throw new PillException(ExceptionMessages.SAVE_ERROR);
+        }
+    }
+
+
+
+    /**
+     * Appends a single item to the storage file.
+     *
+     * @param item The {@link Item} to be saved.
+     * @throws PillException if an error occurs during the saving process.
+     */
+    public void saveItem(Item item) throws PillException {
+        try {
+            File file = initializeFile();
+            FileWriter fw = new FileWriter(file, true);
+            fw.write(item.getName() + SEPARATOR + item.getQuantity());
+            if (item.getExpiryDate().isPresent()) {
+                fw.write(SEPARATOR + item.getExpiryDate().get());
+            }
+            fw.write(System.lineSeparator());
+            fw.close();
+        } catch (IOException e) {
+            throw new PillException(ExceptionMessages.SAVE_ERROR);
+        }
+    }
+
+    /**
+     * Loads saved CSV data into an ItemMap
+     *
+     * @return The ItemMap containing saved items
+     */
+    public ItemMap loadData() {
+        ItemMap loadedItems = new ItemMap();
+        try {
+            File file = initializeFile();
+            Scanner scanner = new Scanner(file);
+            while (scanner.hasNextLine()) {
+                try {
+                    String line = scanner.nextLine();
+                    Item item = loadLine(line);
+                    loadedItems.addItemSilent(item);
+                } catch (PillException e) {
+                    PillException.printException(e);
+                }
+            }
+            scanner.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return loadedItems;
+    }
+
+    /**
+     * Returns data in current line as an Item
+     * @param line Next line string read by the scanner
+     * @return The item present in the line
+     * @throws PillException if format of saved data is incorrect
+     */
+    public Item loadLine(String line) throws PillException {
+        Item item;
+        String[] data = line.split(SEPARATOR);
+        if (data.length == 3) {
+            try {
+                item = new Item(data[0], Integer.parseInt(data[1]), LocalDate.parse(data[2]));
```

Comment:
Maybe you could refactor `data[0]` and so on such that it is more understandable what each variable means before the if statement